### ISSUE TITLE
problems reading /dev/urandom during boot

### ISSUE THREAD

**Issue by smoser** [cirros-dev/cirros#34]
When looking at logs recently ([example](https://travis-ci.com/cirros-dev/cirros/jobs/280515921), I've started to see lines like:

    [   14.578405] random: dropbearkey: uninitialized urandom read (32 bytes read)
    [   14.594791] random: dropbearkey: uninitialized urandom read (32 bytes read)

those maybe could just be considered noise, but then real fallout occurs like:

    WARN: generating key of type dss failed!

If we fail to generate a ssh key, then ssh connection could fail.  Here the
error is a dss key, which is probably not used, but that is besides the point.
Also, I'm aware that dropbear will generate keys on connetion time if they're
not present, in an attempt to have better entropy.  But theres actually no
guarantee that the attempt later would be any better.

I believe that these errors are new, but I do not know how new.  I do not
believe they occur in 0.4.0.

This systemd issue seems related https://github.com/systemd/systemd/issues/4167 .

**Comment by hrw** [comment]
I booted x86-64 image with bin/test-boot. Due to Virtio-rng used there was enough entropy to start:
```
[    2.966691] random: fast init done
[    2.967333] random: crng init done
```

Dropbear generated ecdsa and rsa keys without problems. We do not do dss key anymore now.

**Comment by smoser** [comment]
Well, it appears that the issues are just warnings.
Attached is a log of 

    qemu-system-x86_64 -m 512 -machine pc \
        -device virtio-net-pci,netdev=net00 \
        -netdev type=user,id=net00,net=10.0.12.0/24,host=10.0.12.2 \
        -drive if=virtio,file=seed.img,format=raw \
        -nographic -vga none \
        -kernel cirros-0.5.0_pre4-x86_64-kernel \
        -initrd cirros-0.5.0_pre4-x86_64-initramfs \
        -append 'console=ttyS0'

[cirros-boot-no-hwrandom.log](https://github.com/cirros-dev/cirros/files/4282371/cirros-boot-no-hwrandom.log)

**Comment by smoser** [comment]
I'd love to have some way to avoid those warnings, but they do appear to be just warnings.

The dss key generation error is un-related to the warnings, it was fixed by #46 .
I'll close this issue for now.
